trajopt_ifopt/include/trajopt_ifopt/constraints/collision_constraint.h

```diff
@@ -73,7 +73,7 @@ class CollisionConstraintIfopt : public ifopt::ConstraintSet
    * @brief Fills the jacobian block associated with the constraint
    * @param jac_block Block of the overall jacobian associated with these constraints
    */
-  void CalcJacobianBlock(const Eigen::Ref<Eigen::VectorXd>& joint_vals, Jacobian& jac_block) const;
+  void CalcJacobianBlock(const Eigen::Ref<const Eigen::VectorXd>& joint_vals, Jacobian& jac_block) const;
```

Comment:
Slightly unrelated, but should this be `Eigen::Ref<Jacobian>` ? [This](https://stackoverflow.com/questions/21132538/correct-usage-of-the-eigenref-class) answer indicates that for writable reference, that is preferred.

Reply:
Yep if it is an Eigen Type. I will make the change.

Reply:
Well, it looks like we cannot use the Ref because we are resizing the matrix size.

Reply:
That makes me feel better. I wondered why I didn't do that.